### PR TITLE
Feature: Added a prompt when failing to rename items requiring additional permissions

### LINKED ISSUE #8326: No under permission warning when renaming some specific files
### Description

No under permission warnings are shown when renaming Utilman.exe to Utilman2.exe under system32

### Steps To Reproduce

1. Go to system32
2. find Utilman.exe
2. Rename it to Utilman2.exe
3. Select "Yes" for UAC
4. Nothing changed.

### Expected behavior

There is no rename failure message. Normally it should tell us to get permission from TrustedInstaller

### Files Version

Version: 2.1.13.0

### Windows Version

Windows 11 Dev 22543.1000

### Relevant Assets/Logs

[debug_fulltrust.log](https://github.com/files-community/Files/files/8054901/debug_fulltrust.log)
[debug.log](https://github.com/files-community/Files/files/8054902/debug.log)

2022-02-13

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #8326 

**Known bugs**
- There are two cases where `Edit permissions` doesn't work:
  1. You start to rename an item on one page, and then (somehow) navigate to another before clicking `Edit permissions`
  2. You open the properties window for that item, navigate to a different page and then rename it

The issue behind these bugs is that the `Properties` window depends on `IShellPage` and `ListedItem`, while `RenameAsync` doesn't. To get those values, I'm using `ContentPageContext`, but this doesn't work if you change the working directory while renaming. Do you have any ideas on how I can fix this without messing up the code too much?

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Navigate to `C:\Windows\Ssytem32`
   4. Try to rename any file/folder (such as `Utilman.exe`)
   5. When the warning is shown try clicking both `Ok` and `Edit permissions`

**Screenshots**
Screenshot 2024-02-07 210409

### CODE REVIEW

**Comment by yaira2** [comment]
Trusted Installer is just an example, I imagine there can be others cases as well.

**Comment by ferrariofilippo** [comment]
> Trusted Installer is just an example, I imagine there can be others cases as well.

I guess it can only be either `TrustedInstaller` or `SYSTEM`. Is there any other account with greater privileges than admin?
Anyway, I need to find a way to distinguish the two cases since they both return code `0x80270021` (`COPYENGINE_E_ACCESS_DENIED_SRC`)

**Comment by ferrariofilippo** [comment]
I think we should also extended this to other operations such as `Delete`

**Comment by yaira2** [comment]
> Is there any other account with greater privileges than admin?

It could also be a non admin account

**Comment by ferrariofilippo** [comment]
> It could also be a non admin account that needs access from admin.

In that case there's the UAC modal

**Comment by ferrariofilippo** [comment]
I tried to find a way to distinct between `TrustedInstaller` and `SYSTEM`, but got nowhere.
Using `FileSecurityHelpers.GetAccessControlList()` and `FileSecurityHelpers.GetOwner()`, I can't get anything but generic errors or exceptions (e.g. "Access denied", "NullReference" caused by `source.ItemType` or by `GetOwner()`).
I can either use a more general string (e.g. "requires higher permissions") or try again, but in this case I may need some help

**Comment by yaira2** [comment]
> I can either use a more general string (e.g. "requires higher permissions")

I think that's fine

File: src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs [hunk 1]
```diff
@@ -33,6 +33,16 @@ public MainPropertiesPage()
 				FlowDirection = FlowDirection.RightToLeft;
 		}
 
+		public bool TryNavigateToPage(PropertiesNavigationViewItemType pageType)
```
Comment (yaira2) [inline src/Files.App/Views/Properties/MainPropertiesPage.xaml.cs:36]:
```suggestion
		
		// Navigates to specified properties page
		public bool TryNavigateToPage(PropertiesNavigationViewItemType pageType)
```

**Review by yaira2 — APPROVED** [review]
LGTM

**Review by 0x5bfa — COMMENTED** [review]
LGTM regarding codebase quaility

**Pull request merged**